app/controllers/reviews_controller.rb

```diff
@@ -2,13 +2,48 @@ class ReviewsController < ApplicationController
   def new
     @task = API::Task.includes(:framework).find(params[:task_id]).first
 
-    submission = API::Submission.includes(:entries).find(params[:submission_id]).first
+    @invoices_count = submission_entries.count { |e| e.source['sheet'].match?(/invoice/i) }
+    @orders_count = submission_entries.count { |e| e.source['sheet'].match?(/order/i) }
+
+    @validator = Validator.new(submission_entries: submission_entries)
 
-    @invoices_count = submission.entries.count { |e| e.source['sheet'].match?(/invoice/i) }
-    @orders_count = submission.entries.count { |e| e.source['sheet'].match?(/order/i) }
+    case @validator.status
+    when 'validated'
+      flash.now[:notice] = 'All entries are valid'
+    when 'errored'
+      flash.now[:alert] = "There are #{@validator.errors.count} errors in the uploaded spreadsheet"
+    end
   end
 
   def create
-    # NO OP
+    render 'tasks/complete'
+  end
+
+  def processing
+    redirect_to new_task_submission_review_path if ingest_entries_completed?
+  end
+
+  def ingest_status_polling
+    render json: { complete: ingest_entries_completed? }
+  end
+
+  private
+
+  def submission_entries
+    submission = API::Submission.includes(:entries).find(params[:submission_id]).first
+    submission.entries
+  end
+
+  def submission_file
+    submission_file_id = submission_entries.first.submission_file_id
```

Comment:
When a file is first uploaded, there are no entries yet because it is still ingesting, which causes this is fail with `undefined method `submission_file_id' for nil:NilClass`

Reply:
Also, you can get the submission files' ID through `Submission` (with a minor change to the API submission serializer) - i.e.

```ruby
submission = API::Submission.includes(files: :entries).find(params[:submission_id]).first
submission.files&.first&.id
```

Reply:
@esteedqueen The above needs a change to the API, so how about we just guard against `submission_entries` being `nil` for now..?